controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
+		return configMap, err
+	}
+	configMap.Name = fmt.Sprintf("%s-%s", configMap.Name, h)
+
+	return configMap, nil
+}
+
+/**
+ * This is in its own method because it will involve
+ * reading through multiple levels of the configuration to see
+ * what the engine type is. It may even involve inferring an engine
+ * type based on the stack contents (though that may happen earlier
+ * in the lifecycle than this).
+ */
+func (r *HelmChartInstallReconciler) getEngineType(
+	claim *unstructured.Unstructured,
+	stackConfig *v1alpha1.StackConfiguration,
+) string {
+	return "helm2"
 }
 
-func (r *HelmChartInstallReconciler) processConfiguration(
+func (r *HelmChartInstallReconciler) executeBehavior(
 	ctx context.Context,
-	claim *helmv1alpha1.HelmChartInstall,
-	configuration unstructured.Unstructured,
-) (unstructured.Unstructured, error) {
-	// Given a claim and a stack configuration, transform the claim into something
-	// to inject into a stack's configuration renderer, and then render the Stack's
-	// resources from the configuration and the transformed claim
+	claim *unstructured.Unstructured,
+	engineConfiguration *corev1.ConfigMap,
+	targetStackImage string,
+	behavior *v1alpha1.StackConfigurationBehavior,
+) (*unstructured.Unstructured, error) {
+	for _, resource := range behavior.Resources {
+		// TODO error handling
+		// TODO use result
+		r.executeHook(ctx, claim, engineConfiguration, targetStackImage, resource)
+	}
+	// TODO return a real result
+	return &unstructured.Unstructured{}, nil
+}
+
+// TODO we could have a method create the job, and a higher-level one execute it.
+func (r *HelmChartInstallReconciler) executeHook(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	engineConfig *corev1.ConfigMap,
+	targetStackImage string,
+	targetResourceDir string,
+) (*unstructured.Unstructured, error) {
+	// TODO if there is no config specified, either use an empty config or don't specify
+	// one at all.
+
 	ownerRef := meta.AsOwner(meta.ReferenceTo(claim, claim.GroupVersionKind()))
 	var jobBackoff int32
 	jobBackoff = 0
-	// TODO Handle multiple resource dirs. Perhaps try rolling everything into a single job,
-	//      so have multiple engine containers, one for each path.
-	//      And multiple apply containers; one for each path. Or, make multiple jobs.
-	// TODO we may want some of these steps to be in this controller instead of
-	//      being outsourced
-
-	// TODO resource dir will come from the stack configuration
-	targetResourceDir := "myResourceDir"
-	// TODO target stack image will come from the claim
-	targetStackImage := "crossplane/template-stack:latest"
+
+	// TODO target stack image will come from the stack object, or maybe the stack install object.
+	// Then for each resource behavior hook, we want to run the hook
+	// TODO update this to use the most recent format, where a hook is a structured object
+
 	resourceDirName := fmt.Sprintf("/.registry/resources/%s", targetResourceDir)
```

Comment:
https://golang.org/pkg/path/filepath/#Join would probably make sense here.

Reply:
That's true. When I was creating the arguments for the `cp` command, I ended up using `Sprintf` because `Join` was removing the `/.` suffix that I wanted. And I wasn't sure how I felt about using two different patterns for creating paths, so I went with using the same pattern for all of the path manipulations.

Reply:
I've left this as-is for now.